### PR TITLE
Translatable files

### PR DESCRIPTION
**Pull request by wprod**

### CODE REVIEW

File: src/Services/UploadImageService.php [hunk 1]
```diff
@@ -148,4 +160,64 @@ protected function setUploadedImage(string $imageAttributeName)
...
     }
+
+    /**
+     * Fill Model image attribute with good value associated to the good language
+     *
+     * @param string $imageAttributeName
+     */
+    protected function setUploadedImageLang(string $imageAttributeName)
+    {
+        $locale = $_POST['locale'];
```
Comment (ghost) [inline src/Services/UploadImageService.php]: I does not work if it is executed in command line. For instance a seeder.  
```bash
  [ErrorException]         
  Undefined index: locale  
```
Could we have a default value? something like:
```php
$locale = !empty($_POST['locale']) ? $_POST['locale'] : \App::getLocale();
```

File: src/Services/UploadImageService.php [hunk 2]
```diff
@@ -148,4 +160,64 @@ protected function setUploadedImage(string $imageAttributeName)
...
+        $value = $this->model->getTranslation($imageAttributeName, $locale);
+        $original = $originalLocale = null;
```
Comment (ghost) [inline src/Services/UploadImageService.php]: Unused local variable 'original'. The value of the variable is overwritten immediately

File: src/ModelTraits/UploadableImage.php [hunk 3]
```diff
@@ -35,7 +35,14 @@ public static function bootUploadableImage()
      */
     public function fillUploadedImageAttributeValue(string $imageAttributeName, string $path)
     {
-        $this->{$imageAttributeName} = $path;
+        if (method_exists($this, 'isTranslatableAttribute')
+            && is_callable([$this, 'isTranslatableAttribute'])
+            && $this->isTranslatableAttribute($imageAttributeName)
+        ) {
+            $this->setTranslation($imageAttributeName, request('locale', $default = null), $path);
```
Comment (ghost) [inline src/ModelTraits/UploadableImage.php]: setTranslation method does not accept null as locale.

```bash
Type error: Argument 2 passed to App\Models\Tag::setTranslation() must be of the type string, null given,
```

File: src/Services/UploadFileService.php [hunk 4]
```diff
@@ -110,6 +117,36 @@ public function deleteFiles(Model $model)
     }
 
     /**
+     * Delete old file on disk
+     *
+     * @param $originalAttributeName
+     * @param $fileAttributeName
+     * @return bool
+     */
+    protected function deleteOdlFile($originalAttributeName, $fileAttributeName)
```
Comment (ghost) [inline src/Services/UploadFileService.php]: did you mean deleteOldFile?

File: src/ModelTraits/UploadableFile.php [hunk 5]
```diff
@@ -21,14 +28,21 @@ public static function bootUploadableFile()
...
+     * Put value on desired model document attribute
      *
      * @param string $fileAttributeName
      * @param string $path
      */
     public function fillUploadedFileAttributeValue(string $fileAttributeName, string $path)
     {
-        $this->{$fileAttributeName} = $path;
+        if (method_exists($this, 'isTranslatableAttribute')
+            && is_callable([$this, 'isTranslatableAttribute'])
```
Comment (ghost) [inline src/ModelTraits/UploadableFile.php]:
```php
@if (method_exists($crud->model, 'isTranslatableAttribute') && $crud->model->isTranslatableAttribute($field['name']) && config('backpack.crud.show_translatable_field_icon'))
<i class="fa fa-flag-checkered pull-{{ config('backpack.crud.translatable_field_icon_position') }}" style="margin-top: 3px;" title="This field is translatable."></i>
@endif
```
apparently if this method is protected, the whole application will crash... Backpack expects this method to be public

**Comment by jguyomard** [comment]
I think some refacto are needed for image trait, but i merge anyway, we'll improve this later.

**Comment by jguyomard** [comment]
Thx!

**Pull request merged**